test/TesterInternal/StorageTests/PersistenceGrainTests.cs

```diff
@@ -325,6 +325,26 @@ public void Persistence_Silo_StorageProvider_Name_Missing()
         }
 
         [Fact, TestCategory("Functional"), TestCategory("Persistence")]
+        public async Task Persistence_Read_Write_Struct_State()
+        {
+            const string providerName = "test1";
+            Guid id = Guid.NewGuid();
+            var grain = GrainClient.GrainFactory.GetGrain<IValueTypeTestGrain>(id);
+
+            var initial = await grain.GetStateData();
+            Assert.AreEqual(0, initial);
+
+            await grain.SetStateData(long.MaxValue);
+
+            MockStorageProvider storageProvider = FindStorageProviderInUse(providerName);
+
+            Assert.AreEqual(2, storageProvider.ReadCount, "StorageProvider #Reads");
+            Assert.AreEqual(1, storageProvider.WriteCount, "StorageProvider #Writes");
```

Comment:
I only added the lines to use the MockStorageProvider as that was the closest thing I could find to mirror my test off of.

I'll limit my test to the grain's public interface and go ahead and move this test.

What preference do you have for where this test winds up?